components/raftstore/src/store/peer.rs

```diff
@@ -3371,58 +3366,39 @@ where
             pending_peers: self.collect_pending_peers(ctx),
             written_bytes: self.peer_stat.written_bytes,
             written_keys: self.peer_stat.written_keys,
-            approximate_size: self.approximate_size.unwrap_or_default(),
-            approximate_keys: self.approximate_keys.unwrap_or_default(),
+            approximate_size: self.approximate_size,
+            approximate_keys: self.approximate_keys,
             replication_status: self.region_replication_status(),
         });
-        if !self.is_region_size_or_keys_none() {
-            if let Err(e) = ctx.pd_scheduler.schedule(task) {
-                error!(
-                    "failed to notify pd";
-                    "region_id" => self.region_id,
-                    "peer_id" => self.peer.get_id(),
-                    "err" => ?e,
-                );
-            }
+        if let Err(e) = ctx.pd_scheduler.schedule(task) {
+            error!(
+                "failed to notify pd";
+                "region_id" => self.region_id,
+                "peer_id" => self.peer.get_id(),
+                "err" => ?e,
+            );
             return;
         }
-
-        if self.pending_pd_heartbeat_tasks.load(Ordering::SeqCst) > 2 {
-            return;
+        fail_point!("schedule_check_split");
+        if !self.has_calculated_region_size {
+            if ctx.importer.get_mode() != SwitchMode::Import {
+                self.schedule_check_split(ctx);
```

Comment:
How about waiting for split check tick? If so, the function `schedule_check_split` can be removed and the logic in it can be returned back to `on_split_region_check_tick`